policies/versioning-policy.md

```diff
@@ -0,0 +1,155 @@
+# OpenSSL release versioning policy
+
+This document describes the release versioning scheme used by the OpenSSL
+project from version 3.0.0 onwards.  It also details the level of [ABI] and
+[API] compatibility each version represents.
+
+> _Note: All examples herein are illustrative and do not constitute part of the
+> versioning policy._
+
+The version scheme consists a triple of numbers:
+**[major]**.**[minor]**.**[patch]**.
+
+> For example, the version _3.0.1_ has a _major_ version of _3_, a _minor_
+> version of _0_ and a _patch_ version of _1_.
+
+This closely aligns with the expectations of users who are familiar
+with _semantic versioning_.  However, it is not quite _semantic versioning_
+because adopting _semantic versioning_ would mean changing our current
+Long Term Support ([LTS]) policies and practices.
+
+## Major release: changing the first digit
+
+A major release can, and generally will, introduce API and ABI breaking
+changes.
+
+> For example, a program that runs with OpenSSL 3.0.1 should not expect to run
+> with OpenSSL 4.0.0 without modifications.  If it does run, it might not be
+> as efficient as it was.
+
+In order to maintain stability and limit rework across major versions:
+
+- Existing [public interfaces] will remain unmodified except where changes
+  are unlikely to break source compatibility or where structures are being
+  made opaque.
+- Existing public interfaces must remain in at least one LTS release
+  until the original interface has been documented as _deprecated_
+  for at least 5 years.
+- When structures are made opaque, any newly required accessor macros
+  or functions are added in a feature release of the extant LTS release
+  and all supported intermediate successor releases.
+
+Exceptions to these rules require a vote by the [OMC].  See also the
+[stable release update policy].
+
+## Minor release: changing the second digit
+
+A minor release can, and generally will, introduce new features.  However both
+the API and ABI will be preserved.
+
+> For example, a program built with OpenSSL release 3.0.1 will be able to
+> run with OpenSSL 3.1.0 but might not be able to take advantage of new
+> features without modification.
+
+Exceptions to these rules require a vote by the OMC.
+
+## Patch release: changing the third digit
+
+A patch release will only contain bug fixes.  Both the API and ABI will remain
+compatible across patch releases.
+
+> For example, a program linked with OpenSSL release 3.0.0 can
+> run with OpenSSL 3.0.1 without changes.
+
+Exceptions to these rules require a vote by the OMC.
+
+## Long term stable releases
+
+A release can be designated as a release.  LTS releases will be supported
+for at least five years and the project will specify a LTS release at
+least every four years.  Non-LTS releases will be supported for at least
```

Comment:
Do we need to specify them more frequently than that? After four years the current LTS release will have 0 months remaining of normal bug fixes, and 12 months of security fixes only.

Reply:
The project can make them more frequently, however it must specify one every four years.
Thus, I don't think we want to mandate them being more frequent.